daemon/exec.go

```diff
@@ -183,7 +184,9 @@ func (d *Daemon) ContainerExecStart(execName string, stdin io.ReadCloser, stdout
 		go func() {
 			defer w.Close()
 			defer logrus.Debugf("Closing buffered stdin pipe")
-			io.Copy(w, stdin)
+			buf := pools.BufioReader32KPool.Get(stdin)
+			defer pools.BufioReader32KPool.Put(buf)
+			io.Copy(w, buf)
```

Comment:
I know this is unrelated to the PR, but could we add a check for this error here.